python/graphstorm/gconstruct/transform.py

```diff
@@ -853,13 +855,14 @@ class CustomLabelProcessor:
     test_idx : Numpy array
         The array that contains the index of test data points.
     """
-    def __init__(self, col_name, label_name, task_type,
+    def __init__(self, col_name, label_name, id_col, task_type,
                  train_idx=None, val_idx=None, test_idx=None):
+        self._id_col = id_col
         self._col_name = col_name
         self._label_name = label_name
-        self._train_idx = train_idx
-        self._val_idx = val_idx
-        self._test_idx = test_idx
+        self._train_idx = set(train_idx.tolist()) if train_idx is not None else None
```

Comment:
This is a list of strings?

Reply:
these are node IDs. they can be strings or integers.